server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/task/DataTransferTask.java

```diff
@@ -0,0 +1,391 @@
+                List<String> tableColumns = accessor.listTableColumns(schemaName, tableName).stream()
+                        .map(DBTableColumn::getName).collect(Collectors.toList());
+                tableName2ColumnNames.put(tableName, tableColumns);
+            }
+        } finally {
+            try {
+                connectionSession.expire();
+            } catch (Exception e) {
+                // eat exception
+            }
+        }
+
+        Map<SensitiveColumn, MaskingAlgorithm> sensitiveColumn2Algorithm = maskingService
+                .listColumnsAndMaskingAlgorithm(config.getDatabaseId(), tableName2ColumnNames.keySet());
+        if (sensitiveColumn2Algorithm.isEmpty()) {
+            return;
+        }
+        Map<TableColumn, MaskingAlgorithm> column2Algorithm = sensitiveColumn2Algorithm.keySet().stream()
+                .collect(Collectors.toMap(c -> new TableColumn(c.getTableName(), c.getColumnName()),
+                        sensitiveColumn2Algorithm::get, (c1, c2) -> c1));
+        DataMaskerFactory maskerFactory = new DataMaskerFactory();
+        Map<TableIdentity, Map<String, AbstractDataMasker>> maskConfigMap = new HashMap<>();
+        for (String tableName : tableName2ColumnNames.keySet()) {
+            Map<String, AbstractDataMasker> column2Masker = new HashMap<>();
+            for (String columnName : tableName2ColumnNames.get(tableName)) {
+                MaskingAlgorithm algorithm = column2Algorithm.get(new TableColumn(tableName, columnName));
+                if (Objects.isNull(algorithm)) {
+                    continue;
+                }
+                MaskConfig maskConfig = MaskingAlgorithmUtil.toSingleFieldMaskConfig(algorithm, columnName);
+                AbstractDataMasker masker =
+                        maskerFactory.createDataMasker(MaskValueType.SINGLE_VALUE.name(), maskConfig);
+                column2Masker.put(columnName, masker);
+            }
+            maskConfigMap.put(TableIdentity.of(schemaName, tableName), column2Masker);
+        }
+        config.setMaskConfig(maskConfigMap);
+        /*
+         * set max dump size
+         */
+        if (adapter.getMaxDumpSizeBytes() != null) {
+            config.setMaxDumpSizeBytes(adapter.getMaxDumpSizeBytes());
+        }
+        /*
+         * set cursor fetch size
+         */
+        config.setCursorFetchSize(properties.getCursorFetchSize());
+
+    }
+
+    private void postHandleForExport(DataTransferTaskResult result) throws IOException {
+        if (!workingDir.exists()) {
+            throw new FileNotFoundException(workingDir + " is not found");
+        }
+
+        File exportPath = Paths.get(workingDir.getPath(), "data").toFile();
+        copyExportedFiles(result, exportPath.getPath());
+        File dest = new File(workingDir.getPath() + File.separator + workingDir.getName() + "_export_file.zip");
+        try {
+            DumperOutput output = new DumperOutput(exportPath);
```

Comment:
not a general logic

Reply:
We will copy the output of plugins into a folder which DumperOutput could recognize, so this is a general logic.